src/transcript-plugin.tsx

```diff
@@ -1,401 +1,294 @@
+        // define transcript item group
+        const relevantTranscriptItem = this._data.find(item => item.id === latestTranscriptCuePoint.id);
+        if (relevantTranscriptItem) {
+          this._highlightedGroup = this._data.filter(item => {
+            return item.displayTime === relevantTranscriptItem.displayTime;
+          });
+          if (this._highlightedGroup.length) {
+            this._highlightedGroup.forEach(item => {
+              this._activeCuePointsMap[item.id] = true;
+            });
+            this._updateTranscriptPanel();
           }
-        },
-        err => {
-          this._onError(err, "Failed to fetch captions list", "_getCaptionsList");
         }
-    );
+      }
+    }
   };
 
-  private _filterCaptionAssetsByProperty = (
-    list: KalturaCaptionAsset[],
-    match: string | null,
-    property: string
-  ): KalturaCaptionAsset[] => {
-    return list.filter((kalturaCaptionAsset: KalturaCaptionAsset) => {
-      return get(kalturaCaptionAsset, property, null) === match;
-    });
-  }
+  private _registerCuePointTypes = () => {
+    const cuePointTypes: Array<string> = [];
 
-  private _findCaptionAsset = (
-      event: string | Record<string, any>
-  ): KalturaCaptionAsset => {
-    if (typeof event === 'string') {
-      const filteredByLang = this._filterCaptionAssetsByProperty(this._captionsList, event, 'languageCode');
-      // take first captions from caption-list when caption language is not defined
-      return filteredByLang[0] ? filteredByLang[0] : this._captionsList[0];
+    if (this._itemsFilter[ItemTypes.Caption]) {
+      cuePointTypes.push(this.cuePointManager.CuepointType.CAPTION);
     }
-    const filteredByLang = this._filterCaptionAssetsByProperty(
-      this._captionsList,
-      get(event, 'payload.selectedTextTrack._language', null),
-      'languageCode'
-    );
-    if (filteredByLang.length === 1) {
-      return filteredByLang[0];
+    this.cuePointManager.registerTypes(cuePointTypes);
+  };
+
+  private _addCaptionData = (newData: ItemData[]) => {
+    this._activeCaptionMapId = this._getCaptionMapId();
+    this._captionMap.set(this._activeCaptionMapId, newData);
+    this._createOrUpdatePlugin();
+  };
+
+  private _getCaptionMapId = (): string => {
+    const allTextTracks = this.player.getTracks(this.player.Track.TEXT) || [];
+    const activeTextTrack = allTextTracks.find(track => track.active);
+    if (activeTextTrack) {
+      const captionMapId = `${activeTextTrack.language}-${activeTextTrack.label}`;
+      return captionMapId;
     }
-    const filteredByLabel = this._filterCaptionAssetsByProperty(
-      filteredByLang,
-      get(event, 'payload.selectedTextTrack._label', null),
-      'label'
-    );
-    if (filteredByLang.length === 1) {
-      return filteredByLabel[0];
+    return '';
+  };
+
+  private _getCuePointType = (cue: CuePoint): ItemTypes | null => {
```

Comment:
Since Transcript plugin works only with captions it would better rename method to `isCaptionType` and return boolean result if CuePoint is caption or not